### PR TITLE
fixed profiler dependency on new gftl-v2

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
Fix profiler dependency on gftl-v2
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (affects only documentation or cleanup)

## Checklist:

- [x] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Comment by weiyuan-jiang** [comment]
This PR addresses issue #1334

**Comment by mathomp4** [comment]
Blocking for just a second so I can fix up the branch for release.

**Comment by mathomp4** [comment]
Actually, let me keep blocked as I run a GEOSgcm test.

**Comment by mathomp4** [comment]
Zero-diff and builds.

**Comment by weiyuan-jiang** [comment]
Add label DNA back. I want to understand what happens here

**Comment by weiyuan-jiang** [comment]
I guess I figure out what happened here. The include path can be inherited from dependence.  Since profiler depends on mapl.shared, it inherits its include path. Mapl.shared depends on pflogger which has all gFTL's paths.  In our GEOSgcm, we build and use pFlogger but in UFS, only pFlogger_stub is used.  That is why in UFS, v2 path is missing. This PR should work for UFS, but we may add fake dependence to pfloger_stub to make it really mimic pFlogger.

**Comment by tclune** [comment]
@weiyuan-jiang Well done!

And this speaks directly to why it is so important for us to list all the explicit dependencies of each layer.    Otherwise we get surprising results when unrelated layers are changed elsewhere in the model.

**Comment by mathomp4** [comment]
> I guess I figure out what happened here. The include path can be inherited from dependence. Since profiler depends on mapl.shared, it inherits its include path. Mapl.shared depends on pflogger which has all gFTL's paths. In our GEOSgcm, we build and use pFlogger but in UFS, only pFlogger_stub is used. That is why in UFS, v2 path is missing. This PR should work for UFS, but we may add fake dependence to pfloger_stub to make it really mimic pFlogger.

Dang. Good job figuring this out. CMake is...fun this way!. The question for @tclune is: should @weiyuan-jiang add the fake dependence in pflogger_stub?

**Comment by mathomp4** [comment]
@weiyuan-jiang Okay. I talked with @tclune. I think what you should do is this. In *both* `generic/CMakeLists.txt` and in `profiler/CMakeLists.txt`, we should have:

```
GFTL::gftl-v1 GFTL::gftl-v2
```

The reason is that both are currently using both! You can see all the v2 uses:
```
❯ rg 'template.inc'
generic/VarConnVector.F90
8:#include "vector/template.inc"

generic/VarSpecVector.F90
8:#include "vector/template.inc"

profiler/MeterNodeStack.F90
9:#include "vector/template.inc"
```
but of course they use v1 in many more places!

**Comment by weiyuan-jiang** [comment]
> @weiyuan-jiang Okay. I talked with @tclune. I think what you should do is this. In _both_ `generic/CMakeLists.txt` and in `profiler/CMakeLists.txt`, we should have:
> 
> ```
> GFTL::gftl-v1 GFTL::gftl-v2
> ```
> 
> The reason is that both are currently using both! You can see all the v2 uses:
> 
> ```
> ❯ rg 'template.inc'
> generic/VarConnVector.F90
> 8:#include "vector/template.inc"
> 
> generic/VarSpecVector.F90
> 8:#include "vector/template.inc"
> 
> profiler/MeterNodeStack.F90
> 9:#include "vector/template.inc"
> ```
> 
> but of course they use v1 in many more places!

Changed

**Comment by mathomp4** [comment]
Thanks. I'll do a build-and-run test to be sure all is well.

**Comment by mathomp4** [comment]
Works for me. Approving.

**Review by mathomp4 — APPROVED** [review]

File: generic/CMakeLists.txt [hunk 1]
```diff
@@ -56,7 +56,7 @@ find_package(GFTL_SHARED REQUIRED)
 esma_add_library(${this}
   SRCS ${srcs}
   DEPENDENCIES MAPL.oomph MAPL.shared MAPL.profiler MAPL.base
-               PFLOGGER::pflogger GFTL_SHARED::gftl-shared-v2 GFTL_SHARED::gftl-shared-v1 GFTL::gftl
+               PFLOGGER::pflogger GFTL_SHARED::gftl-shared-v2 GFTL_SHARED::gftl-shared-v1 GFTL::gftl GFTL::gftl-v2 GFTL::gftl-v1
```
Comment (tclune) [inline generic/CMakeLists.txt:59]: We want to be thorough, but not redundant.
```suggestion
               PFLOGGER::pflogger GFTL_SHARED::gftl-shared-v2 GFTL_SHARED::gftl-shared-v1  GFTL::gftl-v2 GFTL::gftl-v1
```

Comment (weiyuan-jiang) [reply generic/CMakeLists.txt]: Your suggestion seems fine to me. Why do you take it back?

**Comment by tclune** [comment]
GFTL::gftl and GFTL::gftl-v1 are the same thing.  So one should not specify them both.    

Someday GFTL::gftl will be an alias for gftl-v2.    It sort of provides a future-safe way of upgrading, except that the Fortran files still need to change for it to help.    Probably a waste of effort on my part.

**Review by mathomp4 — APPROVED** [review]

**Pull request merged**